osu.Game/Screens/OnlinePlay/Components/OnlinePlayBackgroundScreen.cs

```diff
@@ -16,65 +21,57 @@ namespace osu.Game.Screens.OnlinePlay.Components
     public abstract partial class OnlinePlayBackgroundScreen : BackgroundScreen
     {
         private CancellationTokenSource? cancellationSource;
-        private PlaylistItemBackground? background;
+        private Background? lastBackground;
+        private int? beatmapId;
 
         [BackgroundDependencyLoader]
         private void load()
         {
-            switchBackground(new PlaylistItemBackground(playlistItem));
+            loadNewBackground();
         }
 
-        private PlaylistItem? playlistItem;
-
         protected PlaylistItem? PlaylistItem
         {
-            get => playlistItem;
             set
             {
-                if (playlistItem == value)
+                if (beatmapId == value?.Beatmap.OnlineID)
                     return;
 
-                playlistItem = value;
+                beatmapId = value?.Beatmap.OnlineID;
 
-                if (LoadState > LoadState.Ready)
-                    updateBackground();
+                if (LoadState >= LoadState.Ready)
+                    loadNewBackground();
             }
         }
 
-        private void updateBackground()
+        private void loadNewBackground()
         {
-            Schedule(() =>
```

Comment:
Schedule removed, doesn't look required.

Reply:
Seems to have originated in a distant past wherein screens would just arbitrarily bind to parts of the room (https://github.com/ppy/osu/commit/44e157447c7d005079e2ede801ee6ad6fe5a802c), so probably fine to remove